### ISSUE TITLE
Crash refreshing VM scaleset

### ISSUE THREAD

**Issue by cehoffman** [terraform-providers/terraform-provider-azurerm#537]
### Terraform Version

`Terraform v0.10.8` with AzureRM `0.3.2`. This same config works fine with `0.1.7` version of AzureRM.

### Affected Resource(s)

`azurerm_virtual_machine_scale_set`

### Terraform Configuration Files
```hcl
resource "azurerm_virtual_machine_scale_set" "docker" {
  count               = "${var.test ? 0 : 1}"
  name                = "${var.prefix}-quay-builder"
  location            = "${var.region}"
  resource_group_name = "${var.resource_group_name}"
  upgrade_policy_mode = "Manual"

  sku {
    name     = "${var.vm_size}"
    tier     = "${element(split("_", var.vm_size),0)}"
    capacity = "${var.builder_count}"
  }

  os_profile {
    computer_name_prefix = "${var.prefix}-quay-builder-"
    admin_username       = "core"
    admin_password       = ""
    custom_data          = "${base64encode(data.ignition_config.builder.rendered)}"
  }

  os_profile_linux_config {
    disable_password_authentication = true

    ssh_keys {
      path     = "/home/core/.ssh/authorized_keys"
      key_data = "${file(var.public_key)}"
    }
  }

  network_profile {
    name    = "quay-builder-profile"
    primary = true

    ip_configuration {
      name                                   = "quay-builder-ip-configuration"
      subnet_id                              = "${azurerm_subnet.docker.id}"
      load_balancer_backend_address_pool_ids = ["${azurerm_lb_backend_address_pool.lb-pool.id}"]
    }
  }

  storage_profile_os_disk {
    name              = ""
    caching           = "ReadWrite"
    create_option     = "FromImage"
    managed_disk_type = "Standard_LRS"
  }

  storage_profile_image_reference {
    publisher = "CoreOS"
    offer     = "CoreOS"
    sku       = "Stable"
    version   = "latest"
  }
}
```

### Debug Output

https://gist.github.com/cehoffman/724aa034741019012a3805058d188d2a

### Panic Output

https://gist.github.com/cehoffman/724aa034741019012a3805058d188d2a

### Expected Behavior

A change requiring a new resource should have been planned.

### Actual Behavior

Terraform crashed with SIGSEGV

### Steps to Reproduce

1. `terraform plan`

**Comment by steven-burns** [comment]
I too am experiencing this issue with azurerm provider v0.3.2 and Terraform v0.10.8.

**Comment by steven-burns** [comment]
I believe this is related to #491

**Comment by tombuildsstuff** [comment]
Hey @cehoffman @steven-burns 

Thanks for opening this issue :)

This is a duplicate of #481 which has been fixed in #480 - the fix has been merged into master and will be released in the near future; as such I'm going to close this issue for the moment.

Thanks!
